src/lib/util/CategoryUtil.php

```diff
@@ -492,12 +492,13 @@ public static function deleteCategoryByID($cid)
     {
         $em = \ServiceUtil::get('doctrine.entitymanager');
 
-        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryEntity c WHERE c.id = :cid";
+        // delete attributes first to avoid error about foreign key constraints
+        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryAttributeEntity a WHERE a.category = :cid";
         $query = $em->createQuery($dql);
         $query->setParameter('cid', $cid);
         $query->getResult();
 
-        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryAttributeEntity a WHERE a.category = :cid";
+        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryEntity c WHERE c.id = :cid";
```

Comment:
I think the two queries should be in a single transaction. This would fix any ordering issues and will make sure the database remains consistent, even if the second step (deleting the CategoryEntities) fails.